### PR TITLE
Separated old_api_map into two attributes.

### PR DESCRIPTION
**Pull request by popovaan**
Root cause analysis: 
'element_type' attribute should not be stored for Result node in old_api_map, but current implementation does not allow optional attributes.

Solution: 
Split old_api_map into two attributes: old_api_map_element_type and old_api_map_order.

Ticket: 70259

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests
* [x]  Transformation tests
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list - N/A
* [x]  Guide on how to convert the **public** model - N/A
* [x]  User guide update - N/A

### CODE REVIEW

File: inference-engine/src/inference_engine/src/ie_network_reader.cpp [hunk 1]
```diff
@@ -357,54 +357,66 @@ CNNNetwork convert_to_cnnnetwork(std::shared_ptr<ngraph::Function>& function,
...
+                    prepost.input(ov::preprocess::InputInfo(i)
+                                          .tensor(InputTensorInfo().set_element_type(type)));
+                    rtInfo.erase(it_type);
+                }
+                else if (it_order != rtInfo.end()) {
+                    prepost.input(ov::preprocess::InputInfo(i)
+                                          .tensor(InputTensorInfo())
+                                          .preprocess(PreProcessSteps().convert_layout(order)));
+                    rtInfo.erase(it_order);
+                }
```
Comment (ilya-lavrenov) [inline inference-engine/src/inference_engine/src/ie_network_reader.cpp:402]: Please, use more convenient syntax from https://github.com/openvinotoolkit/openvino/pull/8629
Now we don't have a lot of if / else

CC @nosovmik

File: inference-engine/src/transformations/include/transformations/rt_info/old_api_map_element_type_attribute.hpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+namespace ov {
+/**
+ * @ingroup ie_runtime_attr_api
+ * @brief OldApiMap class represents runtime info attribute that stores legacy type
+ * that is required for obtaining IR in old API.
+ */
+class TRANSFORMATIONS_API OldApiMapElementType : public VariantImpl<ngraph::element::Type> {
+public:
+OPENVINO_RTTI("old_api_map_element_type", "0");
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/include/transformations/rt_info/old_api_map_element_type_attribute.hpp:33]: Smth wrong with indentation.

Comment (popovaan) [reply inference-engine/src/transformations/include/transformations/rt_info/old_api_map_element_type_attribute.hpp]: Fixed.

File: inference-engine/src/inference_engine/src/ie_network_reader.cpp (hunk 1 above)
Comment (popovaan) [reply inference-engine/src/inference_engine/src/ie_network_reader.cpp:388]: Done.

File: inference-engine/src/transformations/include/transformations/rt_info/old_api_map_element_type_attribute.hpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+#include <openvino/core/rtti.hpp>
+#include <set>
+#include <string>
+#include <transformations_visibility.hpp>
+#include <utility>
+
+namespace ov {
+/**
+ * @ingroup ie_runtime_attr_api
+ * @brief OldApiMap class represents runtime info attribute that stores legacy type
```
Comment (vgavrilo) [inline inference-engine/src/transformations/include/transformations/rt_info/old_api_map_element_type_attribute.hpp:28]: Maybe `@brief OldApiMapElementType class represents...`?

**Review by vgavrilo — CHANGES_REQUESTED** [review]

File: inference-engine/src/transformations/include/transformations/rt_info/old_api_map_element_type_attribute.hpp (hunk 3 above)
Comment (popovaan) [reply inference-engine/src/transformations/include/transformations/rt_info/old_api_map_element_type_attribute.hpp]: Corrected.

File: inference-engine/src/transformations/include/transformations/rt_info/old_api_map_order_attribute.hpp [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        return m_order;
+    }
+};
+
+/**
+ * @ingroup ie_runtime_attr_api
+ * @brief OldApiMapOrder class represents runtime info attribute that stores
+ * order of the transpose that is required for obtaining IR in old API.
+ */
+class TRANSFORMATIONS_API OldApiMapOrder : public VariantImpl<OldApiMapOrderAttr> {
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/include/transformations/rt_info/old_api_map_order_attribute.hpp:74]: Please try just 
```suggestion
class TRANSFORMATIONS_API OldApiMapOrder : public VariantImpl<std::vector<uint64_t>> {
```

**Review by GlebKazantaev — APPROVED** [review]
nGraph changes looks good.

File: inference-engine/src/transformations/include/transformations/rt_info/old_api_map_order_attribute.hpp (hunk 4 above)
Comment (popovaan) [reply inference-engine/src/transformations/include/transformations/rt_info/old_api_map_order_attribute.hpp]: Done.

**Review by vgavrilo — APPROVED** [review]

File: inference-engine/src/transformations/src/transformations/common_optimizations/compress_float_constants.cpp [hunk 5]
```diff
@@ -99,21 +99,7 @@ ov::pass::AddOldApiMapToParameters::AddOldApiMapToParameters() {
             return false;
         auto p_type = param_node->get_element_type();
         if (p_type == ov::element::f32 || p_type == ov::element::f64) {
-            std::vector<uint64_t> order;
-            if (ov::has_old_api_map(node)) {
```
Comment (sadolini) [inline inference-engine/src/transformations/src/transformations/common_optimizations/compress_float_constants.cpp:103]: Why you don't add order for Parameter?

**Review by sadolini — APPROVED** [review]
MO part looks good to me

File: inference-engine/src/transformations/src/transformations/common_optimizations/compress_float_constants.cpp (hunk 5 above)
Comment (popovaan) [reply inference-engine/src/transformations/src/transformations/common_optimizations/compress_float_constants.cpp:103]: Order is not changed in this part. It sets only Parameter type and code below formed default transpose order with [0 .. rank-1] values. As attributes are separated now it is not needed anymore.

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**